x/wasm/internal/types/types.go

```diff
@@ -219,20 +219,21 @@ func NewWasmCoins(cosmosCoins sdk.Coins) (wasmCoins []wasmvmtypes.Coin) {
 const CustomEventType = "wasm"
 const AttributeKeyContractAddr = "contract_address"
 
-// ParseEvents converts wasm LogAttributes into an sdk.Events (with 0 or 1 elements)
-func ParseEvents(logs []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress) sdk.Events {
-	if len(logs) == 0 {
-		return nil
-	}
+// ParseEvents converts wasm LogAttributes into an sdk.Events
+func ParseEvents(wasmOutputAttrs []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress) sdk.Events {
 	// we always tag with the contract address issuing this event
 	attrs := []sdk.Attribute{sdk.NewAttribute(AttributeKeyContractAddr, contractAddr.String())}
-	for _, l := range logs {
+	
+	// append attributes from wasm to the sdk.Event
+	for _, l := range wasmOutputAttrs {
 		// and reserve the contract_address key for our use (not contract)
 		if l.Key != AttributeKeyContractAddr {
 			attr := sdk.NewAttribute(l.Key, l.Value)
 			attrs = append(attrs, attr)
 		}
 	}
+	
+	// each wasm invokation always returns one sdk.Event
```

Comment:
This then seems like a TODO to return `sdk.Event` here.

Reply:
It's semantics IMO, because we'd have to wrap it within an array right after it returns:
- https://github.com/CosmWasm/wasmd/blob/b0b407d5f839cfa1c8f0bc7fce06032a3ca5c774/x/wasm/internal/keeper/keeper.go#L254-L255
- https://github.com/CosmWasm/wasmd/blob/b0b407d5f839cfa1c8f0bc7fce06032a3ca5c774/x/wasm/internal/keeper/keeper.go#L310-L311

Reply:
There is an `EmitEvent` call. But fair enough, leaving it as is for now.